### PR TITLE
Zap Security Scan in the pipeline

### PR DESCRIPTION
**Pull request by jasonlutz-contino**
Implements the ZAP security tool to scan and endpoint and produce a report.

https://www.zaproxy.org/

The pipeline step saves the report as a "github pipeline artifact" called "security-report" eg here https://github.com/contino/gsd-hello-world/actions/runs/2341164629

The actual command is just the `zap-baseline.py` which is pointed at the app running locally.

Compose handles the service dependency and waits for the app to be up before doing the scan.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -41,16 +42,22 @@ verify:
...
 create_table: envfile
 	echo "from create_table"
 	echo "FOO=${FOO}"
 	echo "BAR=${BAR}"
 	${AWS_CLI_DOCKER_COMPOSE} dynamodb create-table \
 	--table-name ${DYNAMODB_TABLE} \
-	--attribute-definitions AttributeName=GIT_COMMIT,AttributeType=S AttributeName=PIPELINE_ID,AttributeType=S \
-	--key-schema AttributeName=GIT_COMMIT,KeyType=HASH AttributeName=PIPELINE_ID,KeyType=RANGE \
-	--provisioned-throughput ReadCapacityUnits=10,WriteCapacityUnits=5
+	--attribute-definitions AttributeName=GIT_COMMIT,AttributeType=S  \
```
Comment (drewkhoury) [inline Makefile:58]: need to figure out why pipeline_id was removed

Comment (drewkhoury) [inline Makefile:58]: need to figure out why pipeline_id was removed

**Comment by drewkhoury** [comment]
Example output:

Screen Shot 2022-05-17 at 3 39 01 PM

Screen Shot 2022-05-17 at 3 39 17 PM

**Comment by drewkhoury** [comment]
Recent changes:

- Created a "zap security" pipeline to make it more obvious/specific what we're doing at this stage.
- Removed the quotes around the port number - just to avoid confusion.
- Added healthcheck and depends-on for better compose-up experience (it was hanging before we added this)
- Renamed one of the unrelated security pipelines to "playground" to make it clear it's not ready for prime-time yet

**Review by pkordes — APPROVED** [review]
lgtm

**Pull request merged**